src/java/org/apache/cassandra/service/accord/AccordCommandStore.java

```diff
@@ -93,8 +94,12 @@ public class AccordCommandStore extends CommandStore implements CacheSize
 {
     private static final Logger logger = LoggerFactory.getLogger(AccordCommandStore.class);
 
+    private static final boolean CHECK_THREADS = CassandraRelevantProperties.TEST_ACCORD_STORE_THREAD_CHECKS_ENABLED.getBoolean();
```

Comment:
Is it beneficial to optional disable this?

We should probably optimise this to not need to submit a runnable, but that is quite easy.

Reply:
I mean that the `Executor` could set it on startup, and expose it as a final field

Reply:
Regarding disabling the checks, might be nice to only disable for notInStore if that's what we care about? Or maybe there's some other cleaner way that doesn't affect other paths.